addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoDeviceHandler.java

```diff
@@ -79,19 +79,24 @@ public void initialize() {
     }
 
     private void scheduleRefreshJob() {
-        logger.debug("Scheduling update channel thread in {} s", refreshStrategy.nextRunDelayInS());
-        refreshJob = scheduler.schedule(() -> {
+        long delay = refreshStrategy.nextRunDelayInS();
+        logger.debug("Scheduling update channel thread in {} s", delay);
+        refreshJob = scheduler.scheduleWithFixedDelay(() -> {
             updateChannels();
-            refreshJob.cancel(false);
-            refreshJob = null;
+            if (refreshJob != null && !refreshJob.isCancelled()) {
```

Comment:
What does this warranty ? I agree with your analysis that leads to thinking a potential die at retrofit level but I can not see the way to catch this

Reply:
Ok, with your second push I now have clearer understanding.

Reply:
This is only for NPE safeguard. We currently don't know where the exception really occurs.

And yes, the exception handling in the second commit is the important part.